client/state/sites/selectors/has-site-product-jetpack-stats-pwyw-only.ts

```diff
@@ -0,0 +1,32 @@
+import { camelOrSnakeSlug } from '@automattic/calypso-products';
+import { productHasStats } from 'calypso/blocks/jetpack-benefits/feature-checks';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { AppState } from 'calypso/types';
+
+function hasSiteProductJetpackStatsPWYWOnly(
+	state: AppState,
+	siteId = getSelectedSiteId( state )
+): boolean {
+	const sitePurchases = getSitePurchases( state, siteId );
+
+	// Get listing of all plans that support stats in some way and qualify as "paid" plans.
+	const plansSupportingStats = sitePurchases?.filter(
+		( product ) =>
+			productHasStats( camelOrSnakeSlug( product ), true ) && product.expiryStatus !== 'expired'
```

Comment:
Side note: We currently only have two features for Stats, `stats-paid and stats-free`, and we probably want to introduce another one `stats-commercial` which includes support for all advanced features and commercial use.